src/query/service/src/interpreters/interpreter_copy.rs

```diff
@@ -454,8 +406,53 @@ impl Interpreter for CopyInterpreter {
         }
 
         match &self.plan {
-            CopyPlan::IntoTable(plan) => self.build_copy_into_table_pipeline(plan).await,
-
+            CopyPlan::IntoTable(plan) => {
+                if plan.enable_distributed {
+                    let distributed_plan_op = self.transform_copy_plan_distributed(plan).await?;
+                    if distributed_plan_op.is_none() {
+                        return self.build_copy_into_table_pipeline(plan).await;
+                    }
+                    let distributed_plan = distributed_plan_op.unwrap();
+
+                    // add exchange plan node to enable distributed
+                    // TODO(leiysky): we reuse the id of exchange here,
+                    // which is not correct. We should generate a new id for insert.
+                    let exchange_plan = PhysicalPlan::Exchange(Exchange {
+                        plan_id: 0,
+                        input: Box::new(PhysicalPlan::DistributedCopyIntoTable(Box::new(
+                            distributed_plan.clone(),
+                        ))),
+                        kind: FragmentKind::Merge,
+                        keys: Vec::new(),
+                    });
+                    let mut build_res =
+                        build_distributed_pipeline(&self.ctx, &exchange_plan, false).await?;
+
+                    let catalog = self.ctx.get_catalog(&distributed_plan.catalog_name)?;
+                    let to_table = catalog.get_table_by_info(&distributed_plan.table_info)?;
+                    let copied_files = CopyInterpreter::upsert_copied_files_request(
```

Comment:
Is ` CopyInterpreter::try_purge_files` performed by on_finished each node?

Reply:
https://github.com/datafuselabs/databend/blob/main/src/query/service/src/interpreters/interpreter_copy.rs#L336

Reply:
do we need extract these work in `set_on_finish` disable them on pipeline on each node and call it only once finally?
may also need a test case with a bad CSV and check no file is purged?